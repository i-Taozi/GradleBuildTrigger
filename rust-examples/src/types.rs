///类型
pub fn types() {
    let decimal = 65.4321_f32;
    //不支持隐式转化
    //let integer: u8 = decimal;
    // FIXME ^ Comment out this line

    //显示转化
    let integer = decimal as u8;
    let character = integer as char;

    println!("Casting: {} -> {} -> {}", decimal, integer, character);

    // 1000 already fits in a u16
    println!("1000 as a u16 is: {}", 1000 as u16);

    // 1000 - 256 - 256 - 256 = 232
    // Under the hood, the first 8 least significant bits (LSB) are kept,
    // while the rest towards the most significant bit (MSB) get truncated.
    //println!("1000 as a u8 is : {}", 1000 as u8);
    // -1 + 256 = 255
    println!("  -1 as a u8 is : {}", (-1i8) as u8);
    println!("1000 mod 256 is : {}", 1000 % 256);

    println!(" 128 as a i16 is: {}", 128 as i16);
    //println!(" 128 as a i8 is : {}", 128 as i8);
    //println!("1000 as a u8 is : {}", 1000 as u8);
    //println!(" 232 as a i8 is : {}", 232 as i8);

    literals();
}

fn literals() {
    //类型作为后缀
    let x = 1u8;
    let y = 2u32;
    let z = 3f32;

    //非固定文字，其类型取决于如何使用
    let i = 1;
    let f = 1.0;

    //Returns the size of the pointed-to value in bytes.
    println!("size of `x` in bytes: {}", std::mem::size_of_val(&x));
    println!("size of `y` in bytes: {}", std::mem::size_of_val(&y));
    println!("size of `z` in bytes: {}", std::mem::size_of_val(&z));
    println!("size of `i` in bytes: {}", std::mem::size_of_val(&i));
    println!("size of `f` in bytes: {}", std::mem::size_of_val(&f));
}
