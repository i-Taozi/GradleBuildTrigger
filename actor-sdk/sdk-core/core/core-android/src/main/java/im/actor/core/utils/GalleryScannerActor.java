package im.actor.core.utils;

import android.content.Context;
import android.database.Cursor;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.net.Uri;
import android.os.Environment;
import android.provider.MediaStore;

import java.util.ArrayList;
import java.util.HashMap;
import im.actor.core.viewmodel.GalleryVM;
import im.actor.runtime.Log;
import im.actor.runtime.actors.Actor;
import im.actor.runtime.collections.SparseArray;

public class GalleryScannerActor extends Actor {

    public static final String TAG = "GALLERY_SCANNER";
    public static final int CHECK_NEW_DELAY = 1000;
    public static final int SCAN_DELAY = 100;
    Context context;
    Uri uri;
    Cursor cursor;
    int offset = 0;
    int column_index_data, column_index_folder_name, column_index_date, column_index_id, column_index_folder_id;

    ArrayList<String> listOfAllImages = new ArrayList<>();
    ArrayList<String> newMedia = new ArrayList<>();

    String absolutePathOfImage = null;
    GalleryVM galleryVM;

    private static final int SCAN_COUNT = 10;

    private boolean visible = false;
    private boolean scanned = false;
    private String[] projection;
    private boolean inited = false;
    private Bitmap bitmap;

    public GalleryScannerActor(Context context, GalleryVM galleryVM) {
        this.context = context;
        this.galleryVM = galleryVM;
    }

    @Override
    public void preStart() {
        uri = android.provider.MediaStore.Images.Media.EXTERNAL_CONTENT_URI;
    }

    private void initScan() {
        Log.d(TAG, "init scan");
        projection = new String[]{MediaStore.Images.Media.DATA,
                MediaStore.Images.Media._ID,
                MediaStore.Images.Media.BUCKET_ID,
                MediaStore.Images.Media.BUCKET_DISPLAY_NAME,
                MediaStore.Images.Media.DATE_TAKEN};

        cursor = getQuery();

        if (cursor != null && cursor.getCount() > 0) {
            column_index_data = cursor.getColumnIndexOrThrow(MediaStore.Images.Media.DATA);
            column_index_id = cursor.getColumnIndexOrThrow(MediaStore.Images.Media._ID);
            column_index_folder_id = cursor.getColumnIndexOrThrow(MediaStore.Images.Media.BUCKET_ID);
            column_index_date = cursor.getColumnIndexOrThrow(MediaStore.Images.Media.DATE_TAKEN);
            column_index_folder_name = cursor
                    .getColumnIndexOrThrow(MediaStore.Images.Media.BUCKET_DISPLAY_NAME);


            Log.d(TAG, "init scan ok - starting scan iterations");
            inited = true;
            self().send(new Scan());
        } else {
            Log.d(TAG, "init scan - no media, let's check it in " + CHECK_NEW_DELAY);
            if (visible) {
                schedule(new InitScan(), CHECK_NEW_DELAY);
            }
        }

    }

    private Cursor getQuery() {
        try {
            return context.getContentResolver().query(uri, projection, null,
                    null, MediaStore.Images.Media.DATE_TAKEN + " DESC");
        } catch (Exception e) {
            return null;
        }
    }

    private void scan() {

        Log.d(TAG, "scan");

        for (int i = 0; i <= SCAN_COUNT && cursor.moveToNext(); i++) {
            absolutePathOfImage = cursor.getString(column_index_data);
            listOfAllImages.add(absolutePathOfImage);
        }

        offset += SCAN_COUNT;
        scanned = offset >= cursor.getCount();

        //
        // Update VM after 1st batch, every 100 photos, when scanned to end
        //
        if (offset == SCAN_COUNT || offset % 100 == 0 || scanned) {
            Log.d(TAG, "scan - update vm, offset - " + offset);
            galleryVM.getGalleryMediaPath().change(new ArrayList<String>(listOfAllImages));
        }

        if (visible) {
            if (!scanned) {
                schedule(new Scan(), SCAN_DELAY);
            } else {
                schedule(new CheckNew(), CHECK_NEW_DELAY);
            }
        }
    }

    private void checkNew() {
        Log.d(TAG, "checkNew");
        cursor = getQuery();
        newMedia.clear();
        boolean firstCycle = true;
        while (cursor != null && cursor.moveToNext()) {
            absolutePathOfImage = cursor.getString(column_index_data);
            if (!listOfAllImages.contains(absolutePathOfImage)) {
                if (firstCycle) {
                    firstCycle = false;
                    bitmap = BitmapFactory.decodeFile(absolutePathOfImage);
                    if (bitmap != null) {
                        bitmap.recycle();
                        newMedia.add(absolutePathOfImage);
                        Log.d(TAG, "checkNew - new media");

                    } else {
                        Log.d(TAG, "checkNew - new media writing, breaking for wait");

                        break;
                    }
                } else {
                    newMedia.add(absolutePathOfImage);
                }

            } else {
                Log.d(TAG, "checkNew - found old media, break");

                break;
            }
        }
        if (newMedia.size() > 0) {
            listOfAllImages.addAll(0, newMedia);
            galleryVM.getGalleryMediaPath().change(new ArrayList<>(listOfAllImages));
            Log.d(TAG, "checkNew - new media add - " + newMedia.size());
        }
        if (visible) {
            Log.d(TAG, "checkNew - visible, schedule next check in " + CHECK_NEW_DELAY);
            schedule(new CheckNew(), CHECK_NEW_DELAY);
        }
    }

    @Override
    public void onReceive(Object message) {
        if (message instanceof Scan) {
            scan();
        } else if (message instanceof InitScan) {
            initScan();
        } else if (message instanceof CheckNew) {
            checkNew();
        } else if (message instanceof Show) {
            visible = true;
            if (scanned) {
                self().send(new CheckNew());
            } else {
                if (!inited) {
                    self().send(new InitScan());
                }
            }
        } else if (message instanceof Hide) {
            visible = false;
        }
    }

    private static class Scan {

    }

    private static class InitScan {

    }

    private static class CheckNew {

    }

    public static class Show {

    }

    public static class Hide {

    }
}
