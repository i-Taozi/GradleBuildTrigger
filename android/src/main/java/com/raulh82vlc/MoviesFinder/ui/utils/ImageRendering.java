package com.raulh82vlc.MoviesFinder.ui.utils;

import android.widget.ImageView;

/**
 * Image Rendering contract
 */
public interface ImageRendering {
    /**
     * load an image
     *
     * @param urlPath path of the image from the API
     * @param img     ImageView where to render
     */
    void loadImage(String urlPath, ImageView img);
}
