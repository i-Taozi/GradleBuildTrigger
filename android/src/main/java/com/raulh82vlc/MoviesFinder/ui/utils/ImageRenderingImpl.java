/*
 * Copyright (C) 2016 Raul Hernandez Lopez
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.raulh82vlc.MoviesFinder.ui.utils;

import android.app.Activity;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.RequestManager;
import com.bumptech.glide.load.engine.DiskCacheStrategy;

import javax.inject.Inject;

/**
 * Image Rendering
 *
 * @author Raul Hernandez Lopez
 */
public class ImageRenderingImpl implements ImageRendering {

    private RequestManager mGlideInstance;

    @Inject
    public ImageRenderingImpl(Activity context) {
        // this might not lead to a null pointer exception but just in case re-initialisation
        if (mGlideInstance == null) {
            synchronized (this) {
                if (mGlideInstance == null) {
                    mGlideInstance = Glide.with(context);
                }
            }
        }
    }

    @Override
    public void loadImage(String urlPath, ImageView img) {
        mGlideInstance
                .load(urlPath)
                .diskCacheStrategy(DiskCacheStrategy.ALL)
                .fitCenter()
                .into(img);
    }
}
