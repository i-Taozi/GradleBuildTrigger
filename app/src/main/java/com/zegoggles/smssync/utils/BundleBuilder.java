package com.zegoggles.smssync.utils;

import android.os.Bundle;
import android.os.Parcelable;

import java.util.ArrayList;

/**
 * A Bundle that doesn't suck. Allows you to chain method calls as you'd expect.
 * https://gist.github.com/felipecsl/45bc010713c0f543354e
 *
 * MIT license
 */
public class BundleBuilder {
    private final Bundle bundle;

    public BundleBuilder() {
        this(new Bundle());
    }

    public BundleBuilder(Bundle existingBundle) {
        bundle = new Bundle(existingBundle);
    }

    public BundleBuilder putBoolean(String key, boolean value) {
        bundle.putBoolean(key, value);
        return this;
    }

    public BundleBuilder putByte(String key, byte value) {
        bundle.putByte(key, value);
        return this;
    }

    public BundleBuilder putChar(String key, char value) {
        bundle.putChar(key, value);
        return this;
    }

    public BundleBuilder putShort(String key, short value) {
        bundle.putShort(key, value);
        return this;
    }

    public BundleBuilder putInt(String key, int value) {
        bundle.putInt(key, value);
        return this;
    }

    public BundleBuilder putLong(String key, long value) {
        bundle.putLong(key, value);
        return this;
    }

    public BundleBuilder putFloat(String key, float value) {
        bundle.putFloat(key, value);
        return this;
    }

    public BundleBuilder putDouble(String key, double value) {
        bundle.putDouble(key, value);
        return this;
    }

    public BundleBuilder putString(String key, String value) {
        bundle.putString(key, value);
        return this;
    }

    public BundleBuilder putParcelable(String key, Parcelable value) {
        bundle.putParcelable(key, value);
        return this;
    }

    public BundleBuilder putParcelableArray(String key, Parcelable[] value) {
        bundle.putParcelableArray(key, value);
        return this;
    }

    public BundleBuilder putParcelableArrayList(String key, ArrayList<? extends Parcelable> value) {
        bundle.putParcelableArrayList(key, value);
        return this;
    }

    public BundleBuilder putAll(Bundle bundle) {
        this.bundle.putAll(bundle);
        return this;
    }

    /**
     * Converts this BundleBuilder into a plain Bundle object.
     *
     * @return A new Bundle containing all the mappings from the current BundleBuilder.
     */
    public Bundle build() {
        return new Bundle(bundle);
    }
}
