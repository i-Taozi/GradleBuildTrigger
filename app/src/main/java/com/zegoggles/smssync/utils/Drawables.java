package com.zegoggles.smssync.utils;

import android.content.res.Resources;
import android.graphics.drawable.Drawable;
import androidx.annotation.NonNull;
import androidx.core.content.res.ResourcesCompat;
import androidx.core.graphics.drawable.DrawableCompat;

public class Drawables {
    private Drawables() {}

    @NonNull
    public static Drawable getTinted(Resources resources, int resource, int color) {
        Drawable drawable = ResourcesCompat.getDrawable(resources, resource, null);
        drawable = DrawableCompat.wrap(drawable);
        DrawableCompat.setTint(drawable.mutate(), color);
        return drawable;
    }
}
