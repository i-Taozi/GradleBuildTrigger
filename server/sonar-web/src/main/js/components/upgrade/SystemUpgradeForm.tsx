/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { filter, flatMap, isEmpty, negate } from 'lodash';
import * as React from 'react';
import { translate } from '../../helpers/l10n';
import { EditionKey } from '../../types/editions';
import { SystemUpgrade } from '../../types/system';
import { ResetButtonLink } from '../controls/buttons';
import Modal from '../controls/Modal';
import { withAppState } from '../hoc/withAppState';
import { Alert, AlertVariant } from '../ui/Alert';
import SystemUpgradeItem from './SystemUpgradeItem';
import { UpdateUseCase } from './utils';

interface Props {
  appState: Pick<T.AppState, 'edition' | 'version'>;
  onClose: () => void;
  systemUpgrades: SystemUpgrade[][];
  latestLTS: string;
  updateUseCase?: UpdateUseCase;
}

const MAP_ALERT: { [key in UpdateUseCase]?: AlertVariant } = {
  [UpdateUseCase.NewPatch]: 'warning',
  [UpdateUseCase.PreLTS]: 'warning',
  [UpdateUseCase.PreviousLTS]: 'error'
};

interface State {
  upgrading: boolean;
}

export class SystemUpgradeForm extends React.PureComponent<Props, State> {
  versionParser = /^(\d+)\.(\d+)(\.(\d+))?/;
  state: State = { upgrading: false };

  render() {
    const { upgrading } = this.state;
    const { appState, systemUpgrades, latestLTS, updateUseCase } = this.props;
    let systemUpgradesWithPatch: SystemUpgrade[][] = [];
    const alertVariant = updateUseCase ? MAP_ALERT[updateUseCase] : undefined;
    const header = translate('system.system_upgrade');
    const parsedVersion = this.versionParser.exec(appState.version);
    let patches: SystemUpgrade[] = [];
    if (updateUseCase === UpdateUseCase.NewPatch && parsedVersion !== null) {
      const [, major, minor] = parsedVersion;
      const majoMinorVersion = `${major}.${minor}`;
      patches = flatMap(systemUpgrades, upgrades =>
        filter(upgrades, upgrade => upgrade.version.startsWith(majoMinorVersion))
      );
      systemUpgradesWithPatch = systemUpgrades
        .map(upgrades => upgrades.filter(upgrade => !upgrade.version.startsWith(majoMinorVersion)))
        .filter(negate(isEmpty));
      systemUpgradesWithPatch.push(patches);
    } else {
      let untilLTS = false;
      for (const upgrades of systemUpgrades) {
        if (untilLTS === false) {
          systemUpgradesWithPatch.push(upgrades);
          untilLTS = upgrades.some(upgrade => upgrade.version.startsWith(latestLTS));
        }
      }
    }

    return (
      <Modal contentLabel={header} onRequestClose={this.props.onClose}>
        <div className="modal-head">
          <h2>{header}</h2>
        </div>

        <div className="modal-body">
          {alertVariant && updateUseCase && (
            <Alert variant={alertVariant} className={`it__upgrade-alert-${updateUseCase}`}>
              {translate('admin_notification.update', updateUseCase)}
            </Alert>
          )}
          {systemUpgradesWithPatch.map(upgrades => (
            <SystemUpgradeItem
              edition={
                appState.edition as EditionKey /* TODO: Fix once AppState is no longer ambiant. */
              }
              key={upgrades[upgrades.length - 1].version}
              systemUpgrades={upgrades}
              isPatch={upgrades === patches}
              isLTSVersion={upgrades.some(upgrade => upgrade.version.startsWith(latestLTS))}
            />
          ))}
        </div>
        <div className="modal-foot">
          {upgrading && <i className="spinner spacer-right" />}
          <a
            className="pull-left"
            href="https://www.sonarqube.org/downloads/?referrer=sonarqube"
            rel="noopener noreferrer"
            target="_blank">
            {translate('system.see_sonarqube_downloads')}
          </a>
          <ResetButtonLink onClick={this.props.onClose}>{translate('cancel')}</ResetButtonLink>
        </div>
      </Modal>
    );
  }
}

export default withAppState(SystemUpgradeForm);
