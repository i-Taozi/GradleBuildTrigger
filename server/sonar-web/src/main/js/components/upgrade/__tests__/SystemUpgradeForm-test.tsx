/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow } from 'enzyme';
import * as React from 'react';
import { EditionKey } from '../../../types/editions';
import { SystemUpgradeForm } from '../SystemUpgradeForm';
import { UpdateUseCase } from '../utils';

const UPGRADES = [
  [
    {
      version: '6.4',
      description: 'Version 6.4 description',
      releaseDate: '2017-06-02',
      changeLogUrl: 'changelogurl',
      downloadUrl: 'downloadurl',
      plugins: {}
    },
    {
      version: '6.3',
      description: 'Version 6.3 description',
      releaseDate: '2017-05-02',
      changeLogUrl: 'changelogurl',
      downloadUrl: 'downloadurl',
      plugins: {}
    }
  ],
  [
    {
      version: '5.6.7',
      description: 'Version 5.6.7 description',
      releaseDate: '2017-03-01',
      changeLogUrl: 'changelogurl',
      downloadUrl: 'downloadurl',
      plugins: {}
    },
    {
      version: '5.6.6',
      description: 'Version 5.6.6 description',
      releaseDate: '2017-04-02',
      changeLogUrl: 'changelogurl',
      downloadUrl: 'downloadurl',
      plugins: {}
    },
    {
      version: '5.6.5',
      description: 'Version 5.6.5 description',
      releaseDate: '2017-03-01',
      changeLogUrl: 'changelogurl',
      downloadUrl: 'downloadurl',
      plugins: {}
    }
  ]
];

it.each([...Object.values(UpdateUseCase), undefined])(
  'should display correctly for %s',
  updateUseCase => {
    expect(
      shallow(
        <SystemUpgradeForm
          appState={{ edition: EditionKey.community, version: '5.6.3' }}
          onClose={jest.fn()}
          systemUpgrades={UPGRADES}
          latestLTS="9.1"
          updateUseCase={updateUseCase}
        />
      )
    ).toMatchSnapshot();
  }
);
