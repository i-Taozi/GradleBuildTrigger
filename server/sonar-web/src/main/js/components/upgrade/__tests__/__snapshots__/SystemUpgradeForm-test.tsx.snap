// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display correctly for new_minor_version 1`] = `
<Modal
  contentLabel="system.system_upgrade"
  onRequestClose={[MockFunction]}
>
  <div
    className="modal-head"
  >
    <h2>
      system.system_upgrade
    </h2>
  </div>
  <div
    className="modal-body"
  >
    <SystemUpgradeItem
      edition="community"
      isLTSVersion={false}
      isPatch={false}
      key="6.3"
      systemUpgrades={
        Array [
          Object {
            "changeLogUrl": "changelogurl",
            "description": "Version 6.4 description",
            "downloadUrl": "downloadurl",
            "plugins": Object {},
            "releaseDate": "2017-06-02",
            "version": "6.4",
          },
          Object {
            "changeLogUrl": "changelogurl",
            "description": "Version 6.3 description",
            "downloadUrl": "downloadurl",
            "plugins": Object {},
            "releaseDate": "2017-05-02",
            "version": "6.3",
          },
        ]
      }
    />
    <SystemUpgradeItem
      edition="community"
      isLTSVersion={false}
      isPatch={false}
      key="5.6.5"
      systemUpgrades={
        Array [
          Object {
            "changeLogUrl": "changelogurl",
            "description": "Version 5.6.7 description",
            "downloadUrl": "downloadurl",
            "plugins": Object {},
            "releaseDate": "2017-03-01",
            "version": "5.6.7",
          },
          Object {
            "changeLogUrl": "changelogurl",
            "description": "Version 5.6.6 description",
            "downloadUrl": "downloadurl",
            "plugins": Object {},
            "releaseDate": "2017-04-02",
            "version": "5.6.6",
          },
          Object {
            "changeLogUrl": "changelogurl",
            "description": "Version 5.6.5 description",
            "downloadUrl": "downloadurl",
            "plugins": Object {},
            "releaseDate": "2017-03-01",
            "version": "5.6.5",
          },
        ]
      }
    />
  </div>
  <div
    className="modal-foot"
  >
    <a
      className="pull-left"
      href="https://www.sonarqube.org/downloads/?referrer=sonarqube"
      rel="noopener noreferrer"
      target="_blank"
    >
      system.see_sonarqube_downloads
    </a>
    <ResetButtonLink
      onClick={[MockFunction]}
    >
      cancel
    </ResetButtonLink>
  </div>
</Modal>
`;

exports[`should display correctly for new_patch 1`] = `
<Modal
  contentLabel="system.system_upgrade"
  onRequestClose={[MockFunction]}
>
  <div
    className="modal-head"
  >
    <h2>
      system.system_upgrade
    </h2>
  </div>
  <div
    className="modal-body"
  >
    <Alert
      className="it__upgrade-alert-new_patch"
      variant="warning"
    >
      admin_notification.update.new_patch
    </Alert>
    <SystemUpgradeItem
      edition="community"
      isLTSVersion={false}
      isPatch={false}
      key="6.3"
      systemUpgrades={
        Array [
          Object {
            "changeLogUrl": "changelogurl",
            "description": "Version 6.4 description",
            "downloadUrl": "downloadurl",
            "plugins": Object {},
            "releaseDate": "2017-06-02",
            "version": "6.4",
          },
          Object {
            "changeLogUrl": "changelogurl",
            "description": "Version 6.3 description",
            "downloadUrl": "downloadurl",
            "plugins": Object {},
            "releaseDate": "2017-05-02",
            "version": "6.3",
          },
        ]
      }
    />
    <SystemUpgradeItem
      edition="community"
      isLTSVersion={false}
      isPatch={true}
      key="5.6.5"
      systemUpgrades={
        Array [
          Object {
            "changeLogUrl": "changelogurl",
            "description": "Version 5.6.7 description",
            "downloadUrl": "downloadurl",
            "plugins": Object {},
            "releaseDate": "2017-03-01",
            "version": "5.6.7",
          },
          Object {
            "changeLogUrl": "changelogurl",
            "description": "Version 5.6.6 description",
            "downloadUrl": "downloadurl",
            "plugins": Object {},
            "releaseDate": "2017-04-02",
            "version": "5.6.6",
          },
          Object {
            "changeLogUrl": "changelogurl",
            "description": "Version 5.6.5 description",
            "downloadUrl": "downloadurl",
            "plugins": Object {},
            "releaseDate": "2017-03-01",
            "version": "5.6.5",
          },
        ]
      }
    />
  </div>
  <div
    className="modal-foot"
  >
    <a
      className="pull-left"
      href="https://www.sonarqube.org/downloads/?referrer=sonarqube"
      rel="noopener noreferrer"
      target="_blank"
    >
      system.see_sonarqube_downloads
    </a>
    <ResetButtonLink
      onClick={[MockFunction]}
    >
      cancel
    </ResetButtonLink>
  </div>
</Modal>
`;

exports[`should display correctly for pre_lts 1`] = `
<Modal
  contentLabel="system.system_upgrade"
  onRequestClose={[MockFunction]}
>
  <div
    className="modal-head"
  >
    <h2>
      system.system_upgrade
    </h2>
  </div>
  <div
    className="modal-body"
  >
    <Alert
      className="it__upgrade-alert-pre_lts"
      variant="warning"
    >
      admin_notification.update.pre_lts
    </Alert>
    <SystemUpgradeItem
      edition="community"
      isLTSVersion={false}
      isPatch={false}
      key="6.3"
      systemUpgrades={
        Array [
          Object {
            "changeLogUrl": "changelogurl",
            "description": "Version 6.4 description",
            "downloadUrl": "downloadurl",
            "plugins": Object {},
            "releaseDate": "2017-06-02",
            "version": "6.4",
          },
          Object {
            "changeLogUrl": "changelogurl",
            "description": "Version 6.3 description",
            "downloadUrl": "downloadurl",
            "plugins": Object {},
            "releaseDate": "2017-05-02",
            "version": "6.3",
          },
        ]
      }
    />
    <SystemUpgradeItem
      edition="community"
      isLTSVersion={false}
      isPatch={false}
      key="5.6.5"
      systemUpgrades={
        Array [
          Object {
            "changeLogUrl": "changelogurl",
            "description": "Version 5.6.7 description",
            "downloadUrl": "downloadurl",
            "plugins": Object {},
            "releaseDate": "2017-03-01",
            "version": "5.6.7",
          },
          Object {
            "changeLogUrl": "changelogurl",
            "description": "Version 5.6.6 description",
            "downloadUrl": "downloadurl",
            "plugins": Object {},
            "releaseDate": "2017-04-02",
            "version": "5.6.6",
          },
          Object {
            "changeLogUrl": "changelogurl",
            "description": "Version 5.6.5 description",
            "downloadUrl": "downloadurl",
            "plugins": Object {},
            "releaseDate": "2017-03-01",
            "version": "5.6.5",
          },
        ]
      }
    />
  </div>
  <div
    className="modal-foot"
  >
    <a
      className="pull-left"
      href="https://www.sonarqube.org/downloads/?referrer=sonarqube"
      rel="noopener noreferrer"
      target="_blank"
    >
      system.see_sonarqube_downloads
    </a>
    <ResetButtonLink
      onClick={[MockFunction]}
    >
      cancel
    </ResetButtonLink>
  </div>
</Modal>
`;

exports[`should display correctly for previous_lts 1`] = `
<Modal
  contentLabel="system.system_upgrade"
  onRequestClose={[MockFunction]}
>
  <div
    className="modal-head"
  >
    <h2>
      system.system_upgrade
    </h2>
  </div>
  <div
    className="modal-body"
  >
    <Alert
      className="it__upgrade-alert-previous_lts"
      variant="error"
    >
      admin_notification.update.previous_lts
    </Alert>
    <SystemUpgradeItem
      edition="community"
      isLTSVersion={false}
      isPatch={false}
      key="6.3"
      systemUpgrades={
        Array [
          Object {
            "changeLogUrl": "changelogurl",
            "description": "Version 6.4 description",
            "downloadUrl": "downloadurl",
            "plugins": Object {},
            "releaseDate": "2017-06-02",
            "version": "6.4",
          },
          Object {
            "changeLogUrl": "changelogurl",
            "description": "Version 6.3 description",
            "downloadUrl": "downloadurl",
            "plugins": Object {},
            "releaseDate": "2017-05-02",
            "version": "6.3",
          },
        ]
      }
    />
    <SystemUpgradeItem
      edition="community"
      isLTSVersion={false}
      isPatch={false}
      key="5.6.5"
      systemUpgrades={
        Array [
          Object {
            "changeLogUrl": "changelogurl",
            "description": "Version 5.6.7 description",
            "downloadUrl": "downloadurl",
            "plugins": Object {},
            "releaseDate": "2017-03-01",
            "version": "5.6.7",
          },
          Object {
            "changeLogUrl": "changelogurl",
            "description": "Version 5.6.6 description",
            "downloadUrl": "downloadurl",
            "plugins": Object {},
            "releaseDate": "2017-04-02",
            "version": "5.6.6",
          },
          Object {
            "changeLogUrl": "changelogurl",
            "description": "Version 5.6.5 description",
            "downloadUrl": "downloadurl",
            "plugins": Object {},
            "releaseDate": "2017-03-01",
            "version": "5.6.5",
          },
        ]
      }
    />
  </div>
  <div
    className="modal-foot"
  >
    <a
      className="pull-left"
      href="https://www.sonarqube.org/downloads/?referrer=sonarqube"
      rel="noopener noreferrer"
      target="_blank"
    >
      system.see_sonarqube_downloads
    </a>
    <ResetButtonLink
      onClick={[MockFunction]}
    >
      cancel
    </ResetButtonLink>
  </div>
</Modal>
`;

exports[`should display correctly for undefined 1`] = `
<Modal
  contentLabel="system.system_upgrade"
  onRequestClose={[MockFunction]}
>
  <div
    className="modal-head"
  >
    <h2>
      system.system_upgrade
    </h2>
  </div>
  <div
    className="modal-body"
  >
    <SystemUpgradeItem
      edition="community"
      isLTSVersion={false}
      isPatch={false}
      key="6.3"
      systemUpgrades={
        Array [
          Object {
            "changeLogUrl": "changelogurl",
            "description": "Version 6.4 description",
            "downloadUrl": "downloadurl",
            "plugins": Object {},
            "releaseDate": "2017-06-02",
            "version": "6.4",
          },
          Object {
            "changeLogUrl": "changelogurl",
            "description": "Version 6.3 description",
            "downloadUrl": "downloadurl",
            "plugins": Object {},
            "releaseDate": "2017-05-02",
            "version": "6.3",
          },
        ]
      }
    />
    <SystemUpgradeItem
      edition="community"
      isLTSVersion={false}
      isPatch={false}
      key="5.6.5"
      systemUpgrades={
        Array [
          Object {
            "changeLogUrl": "changelogurl",
            "description": "Version 5.6.7 description",
            "downloadUrl": "downloadurl",
            "plugins": Object {},
            "releaseDate": "2017-03-01",
            "version": "5.6.7",
          },
          Object {
            "changeLogUrl": "changelogurl",
            "description": "Version 5.6.6 description",
            "downloadUrl": "downloadurl",
            "plugins": Object {},
            "releaseDate": "2017-04-02",
            "version": "5.6.6",
          },
          Object {
            "changeLogUrl": "changelogurl",
            "description": "Version 5.6.5 description",
            "downloadUrl": "downloadurl",
            "plugins": Object {},
            "releaseDate": "2017-03-01",
            "version": "5.6.5",
          },
        ]
      }
    />
  </div>
  <div
    className="modal-foot"
  >
    <a
      className="pull-left"
      href="https://www.sonarqube.org/downloads/?referrer=sonarqube"
      rel="noopener noreferrer"
      target="_blank"
    >
      system.see_sonarqube_downloads
    </a>
    <ResetButtonLink
      onClick={[MockFunction]}
    >
      cancel
    </ResetButtonLink>
  </div>
</Modal>
`;
